aptos-move/framework/move-stdlib/sources/vector.move

```diff
@@ -211,6 +211,21 @@ module std::vector {
         pragma intrinsic = true;
     }
 
+    /// Push a new element to the back of the vector if it is not already in the vector.
+    /// This is O(n) in the worst case and preserves ordering of elements in the vector.
+    /// Return `true` if `key` did not already exist in the vector and `false` otherwise.
+    public fun push_unique<Element: drop>(v: &mut vector<Element>, e: Element): bool {
```

Comment:
My concern with this is that if used improperly, you could end up in a case where you have an inconsistent array.

I'd much rather suggest someone use a well-typed variant of this that doesn't allow for non-unique pushes.

Reply:
What do you mean by inconsistent? For example duplicate items by virtue of using `push_unique` mixed with `push` / `insert` / etc?

If by a well typed variant you mean a separate module, then I have previously implemented and proposed this and it has been widely knocked back: https://aptos-org.slack.com/archives/C036X27DZNG/p1690572419217229.